pkg/service/service.go

```diff
@@ -309,13 +445,66 @@ func (s *Service) InitMaps(ipv6, ipv4, sockMaps, restore bool) error {
 	return nil
 }
 
+func (s *Service) upsertEnvoyEndpoints(svc *svcInfo) error {
+	name := svc.svcNamespace + "/" + svc.svcName
+	lbEndpoints := []*envoy_config_endpoint.LbEndpoint{}
+	for _, be := range svc.backends {
+		lbEndpoints = append(lbEndpoints, &envoy_config_endpoint.LbEndpoint{
```

Comment:
This is getting into "curly bracket hell". 

Can you define a variables for these structures ahead of time and then place the vars in this data structure you're defining?

Reply:
Here the innermost struct carries the port number that is unique for each instance. If we defined this as a variable we'd need to take a deep copy due to every member in the protobuf being a pointer, and then modify the port value inside. Not sure if that would be better?